lib/doc_handler.dart

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+
+library dartpad.doc_handler;
+
+import 'dart:convert';
+import 'dart:html';
+import 'dart:math' as math;
+
+import 'dartservices_client/v1.dart';
+import 'editing/editor.dart';
+import 'context.dart';
+import 'services/common.dart';
+import 'src/ga.dart';
+
+import 'dart_pad.dart';
+import 'package:markd/markdown.dart' as markdown;
+
+class DocHandler {
+  Editor _editor;
+  Context _context;
+  DartservicesApi _dartServices;
+  Analytics ga;
+  static const List cursorKeys = const [KeyCode.LEFT, KeyCode.RIGHT, KeyCode.UP, KeyCode.DOWN];
+
+  final NodeValidatorBuilder _htmlValidator = new NodeValidatorBuilder.common()
+    ..allowElement('a', attributes: ['href'])
+    ..allowElement('img', attributes: ['src']);
+
+  DocHandler(this._editor, this._context, this._dartServices, this.ga) {
+    keys.bind(['f1'], toggleDocTab);
+    document.onClick.listen((e) => _handleClick(e));
```

Comment:
Right now `playground.dart` is the only file that knows the specifics of the UI. I want to keep that the case for now. It'll let us swap out new UIs easily; we'll only have to write a new top-level file (`playground.dart`, `embedded.dart`, ...). So, some aspects of this file should move back into playground.dart; probably anything that knows about a querySelector call or that operates directly on the results of one.